test/core/libp2p/crypto/random_test.cpp

```diff
@@ -51,3 +51,82 @@ TEST(Random, stdGenerator) {
   std::cout << Buffer(bytes2).toHex() << std::endl;
   ASSERT_NE(bytes1, bytes2);
 }
+
+namespace {
+  /**
+   * Taken from
+   * https://github.com/hyperledger/iroha-ed25519/blob/master/test/randombytes/random.cpp#L6
+   * @brief calculates entropy of byte sequence
+   * @param sequence  source byte sequence
+   * @return entropy value
+   * @
+   */
+  double entropy(gsl::span<uint8_t> sequence) {
+    std::vector<uint8_t> freqs(256, 0);
+
+    // calculate frequencies
+    for (unsigned char i : sequence) {
+      ++freqs[i];
+    }
+
+    double e = 0;
+    for (const auto c : freqs) {
+      double freq = (double)c / sequence.size();
+
+      if (freq == 0) {
+        continue;
+      }
+
+      e += freq * log2(freq);
+    }
+
+    return -e;
+  }
+
+  /**
+   * @brief returns max possible entropy for a buffer of given size
+   * @param volume is volume of alphabet
+   * @return max value of entropy for given source alphabet volume
+   */
+  double max_entropy(size_t volume) {
+    return log2(volume);
+  }
+}  // namespace
+
+/**
+ * @given BoostRandomGenerator instance, max entropy value for given source
+ * @when get random bytes and calculate entropy
+ * @then calculated entropy is not less than (max entropy - 2)
+ */
+TEST(Random, enoughEntropyBoost) {
+  BoostRandomGenerator generator;
+  std::vector<uint8_t> buf(256, 0);
+  generator.randomBytes(buf.data(), buf.size());
+
+  auto max = max_entropy(256);  // 8
+  auto ent = entropy(buf);
+
+  std::cout << "max entropy: " << max << std::endl;
+  std::cout << "calculated:  " << ent << std::endl;
+
+  ASSERT_GE(ent, max - 2) << "bad randomness source";
+}
+
+/**
+ * @given StdRandomGenerator instance, max entropy value for given source
+ * @when get random bytes and calculate entropy
+ * @then calculated entropy is not less than (max entropy - 2)
+ */
+TEST(Random, enoughEntropyStd) {
+  StdRandomGenerator generator;
+  std::vector<uint8_t> buf(256, 0);
+  generator.randomBytes(buf.data(), buf.size());
+
+  auto max = max_entropy(256);  // 8
+  auto ent = entropy(buf);
+
+  std::cout << "max entropy: " << max << std::endl;
+  std::cout << "calculated:  " << ent << std::endl;
+
+  ASSERT_GE(ent, max - 2) << "bad randomness source";
+}
```

Comment:
BoostRandomGenerator and StdRandomGenerator have common interface. You can combine tests

Reply:
done